src/cmap/establish/test.rs

```diff
@@ -1,179 +0,0 @@
-use tokio::sync::RwLockWriteGuard;
```

Comment:
This file used to contain unit tests of the speculative authentication behavior, but I had to remove them now that authentication was moved into the `Handshaker` layer, and I couldn't really think of a way to test this without introducing specific logic to make unit testing possible. These being unit tests, they don't use the actual driver and so don't exercise the same code paths a real client would, so it didn't seem like they were providing enough value to be worth preserving. If we do want to unit test this behavior, we can introduce some new functionality to connections for tracking which messages were sent and what not, but I figured we could defer that to later. What do you all think?

Reply:
Not specific to speculative authentication, but our auth test suite should exercise the code paths. We don't have any actual verification that they do, though.